src/coreclr/src/gc/unix/gcenv.unix.cpp

```diff
@@ -478,6 +478,7 @@ uint32_t GCToOSInterface::GetCurrentProcessorNumber()
     assert(processorNumber != -1);
     return processorNumber;
 #else
+    assert(false); // This method is expected to be called only if CanGetCurrentProcessorNumber is true
```

Comment:
Maybe `assert(!CanGetCurrentProcessorNumber())` instead and get rid of the comment?

Reply:
I would then rather add the same assert (`assert(CanGetCurrentProcessorNumber());`) as we have in the other versions of this function at the beginning of the function.

Reply:
I had the assert at the beginning of the function originally. It did not seem to make the intent obvious. That's why I have added the comment instead.

Reply:
I am fine with both ways.